### ISSUE TITLE
`#undef __STDC_IEC_559__` not caught with `-pedanatic-errors`

### ISSUE THREAD

**Issue by dbc7d0cf-60de-4ba2-bc9a-c1b7219315df** [llvm/llvm-project#46140]
|  |  |
| --- | --- |
| Bugzilla Link | [46796](https://llvm.org/bz46796) |
| Version | 10.0 |
| OS | Linux |
| CC | @DougGregor,@zygoloid |

## Extended Description 
The C standard has [in Predefined macro names clause]: None of these macro names ... shall be the subject of `#define` or `#undef` preprocessing directive.

Using clang with `-pedantic-errors` does not catch:
```cpp
 #undef  __STDC_IEC_559__
 #define __STDC_IEC_559__ 1
```

**Comment by Endilll** [comment]
CC @AaronBallman
